contracts/CommonHealthCheck.sol

```diff
@@ -94,37 +73,42 @@ contract CommonHealthCheck {
     }
 
     function check(
+        address strategy,
         uint256 profit,
         uint256 loss,
         uint256 debtPayment,
         uint256 debtOutstanding,
         uint256 totalDebt
     ) external view returns (bool) {
-        return _runChecks(profit, loss, debtPayment, debtOutstanding, totalDebt);
+        require(strategy != address(0));
+
+        return _runChecks(strategy, profit, loss, debtPayment, debtOutstanding, totalDebt);
     }
 
```

Comment:
we can add a new view method called doHealthCheck(address arg)

that returns true by default unless we have a mapping setter for that address turned to true, in the next harvest for that address it can be disabled as a one time harvest free of health checks.

does it make sense?

Reply:
I took a similar approach but used a `mapping(address => bool) public disabledCheck;` so that we don't have to have a structure with a `bool exists;` in order to have it enabled by default.

With that said doHealthCheck looks like this.

```
    function doHealthCheck(address _strategy) {
        return !disabledCheck[_strategy];
    }
```

And I had to add a enableCheck that can only be used by the vault that is attached to the strategy

```
    function enableCheck(address _strategy) external onlyVault(_strategy) {
        disabledCheck[_strategy] = false;
    }

```